ansible/library/announce_routes.py

```diff
@@ -130,12 +136,20 @@ def get_topo_type(topo_name):
     return topo_type
 
 
-def read_topo(topo_name, path):
+def read_topo(topo_name, path, server_index=-1):
     topo_file_path = os.path.join(
         path, TOPO_FILE_FOLDER, TOPO_FILENAME_TEMPLATE.format(topo_name))
     try:
         with open(topo_file_path) as f:
-            return yaml.safe_load(f)
+            _ret = yaml.safe_load(f)
+            if server_index != -1:
+                for _name, _attr in _ret['topology']['VMs'].items():
```

Comment:
So it is a kind of intrusive schema change, is possible to do it smartly? Why user input is required for VM distribution?